src/app/component/CartSummary/CartSummary.componet.js

```diff
@@ -43,11 +48,14 @@ class CartSummary extends Component {
         // eslint-disable-next-line no-param-reassign, no-return-assign
         const itemsTax = items ? items.reduce((sum, { tax_amount }) => sum += tax_amount, tax_amount) : 0;
 
+        const discount = items ? items.reduce(this.calculateDiscount, 0) : 0;
```

Comment:
Is not this field available as a response from server? I believe it is, no need for additional calculation.

Reply:
Feilds "discount_amount" (null) and "subtotal_with_discount" (incorrect) exist on backend, however provided value is incorrect . As for the tax_amount.

Reply:
Actually there are two options: either keep compensating such functionality on FE - in this scenario this PR does exactly that, or to go and implement/fix it in https://github.com/scandipwa/quote-graphql for taxes, discount amount and similar functionality.

Reply:
I prefer the https://github.com/scandipwa/quote-graphql option. If discount or a tax_amount is incorrect, please make sure to fix them! Otherwise, it is completely useless to display those values on FE.

Reply:
I also prefer doing this one on BE and displaying it on FE, but it is not like the values are incorrect on server, but more like https://github.com/scandipwa/quote-graphql/blob/master/src/Model/Resolver/TotalsObject.php#L92